plan/validator.go

```diff
@@ -493,3 +481,100 @@ func isIncorrectName(name string) bool {
 	}
 	return false
 }
+
+func (p *preprocessor) tableUniqueName(schema, table model.CIStr) string {
+	if schema.L != "" && schema.L != p.ctx.GetSessionVars().CurrentDB {
+		return schema.L + "." + table.L
+	}
+	return table.L
+}
+
+func (p *preprocessor) handleTableName(tn *ast.TableName) {
+	if tn.Schema.L == "" {
+		currentDB := p.ctx.GetSessionVars().CurrentDB
+		if currentDB == "" {
+			p.err = errors.Trace(ErrNoDB)
+			return
+		}
+		tn.Schema = model.NewCIStr(currentDB)
+	}
+	if p.inCreateOrDropTable {
+		// The table may not exist in create table or drop table statement.
+		// Skip resolving the table to avoid error.
+		return
+	}
+	if p.inDeleteTableList {
+		idx, ok := p.tableMap[p.tableUniqueName(tn.Schema, tn.Name)]
```

Comment:
The tableMap is empty, so it seems it always returns an error ?

Reply:
DeleteTableList only exists in DeleteStmt,
this commit does not handle DeleteStmt.

SeleteStmt, UpdateStmt, DeleteStmt, UnionStmt will be moved together.

Reply:
So this check can be removed.